lib/vanagon/platform/windows.rb

```diff
@@ -53,9 +54,54 @@ def generate_packaging_artifacts(workdir, name, binding)
       # @param name [String] name of the project
       # @param binding [Binding] binding to use in evaluating the packaging templates
       def generate_msi_packaging_artifacts(workdir, name, binding)
-        FileUtils.mkdir_p(File.join(workdir, "wix"))
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.wxs.erb"), File.join(workdir, "wix",  "#{name}.wxs"), false, { :binding => binding })
-        erb_file(File.join(VANAGON_ROOT, "resources/windows/wix/project.filter.xslt.erb"), File.join(workdir, "wix", "#{name}.filter.xslt"), false, { :binding => binding })
+        # Copy the project specific files first
+        copy_from_project("#{@proj_root}/resources/windows/wix", workdir)
+        merge_defaults_from_vanagon("#{VANAGON_ROOT}/resources/windows/wix", "#{workdir}/wix")
+        process_templates("#{workdir}/wix", binding)
+      end
+
+      # Method to recursively copy from a source project resource directory
+      # to a destination (wix) work directory.
+      # strongly suspect the original cp_r command would have done all of this.
+      #
+      # @param proj_resources [String] Project Resource File directory
+      # @param destination [String] Destination directory
+      # @param verbose [String] True or false
+      def copy_from_project(proj_resources, destination, verbose: false)
+        FileUtils.cp_r(proj_resources, destination, :verbose => verbose)
+      end
+
+      # Method to merge in the files from the Vanagon (generic) directories.
+      # Project Specific files take precedence, so since these are copied prior
+      # to this function, then this merge operation will ignore existing files
+      #
+      # @param vanagon_root [String] Vanagon wix resources directory
+      # @param destination [String] Destination directory
+      # @param verbose [String] True or false
+      def merge_defaults_from_vanagon(vanagon_root, destination, verbose: false)
+        vanagon_path = Pathname.new(vanagon_root)
+        files = Dir.glob(File.join(vanagon_root, "**/*.*"))
+        files.each do |file|
```

Comment:
Technically it's not generating anything, it's simply collecting already generated external objects (the files) in to a list and then acting on the list. The way that it is written at the top level makes sense the way it is for how we want to do this. Specifically I want the function to pass in {VANAGON_ROOT}/resources/windows/wix because the specifications for the WiX filenames and directory structure exist there in the files and not as vanagon objects to be passed in as *files. The way that this is written now makes that clear to me. if we ever moved over to having a list of vanagon objects then we could refactor this, but for now I think this makes more sense.